### ISSUE TITLE
Meilisearch 0.28.0 / meilisearch-rails 0.7.0 API breaking changes

### ISSUE THREAD

**Issue by benbws** [ddnexus/pagy#394]
# Code Issue
- [x] I researched through the [documentation](https://ddnexus.github.io/pagy/), the [pagy issues](https://github.com/ddnexus/pagy/issues) and a known Search Engine, and there is no reference|report|post|video that solves this problem
- [x] I did install|upgraded to the latest version of pagy (or the latest `3.*` for pagy legacy)
- [x] I am providing at least one of the following working and self-contained code support that can reproduce this issue:
   - [x] Edited copy of the single file [pagy_bare_rails.rb](https://github.com/ddnexus/pagy/blob/master/apps/pagy_bare_rails.rb)

### Ruby|Rails|Sinatra|... Versions

Tested on '~> 6.1' but will no doubt apply to all versions as this looks to be a specific pagy meilisearch extension issue.

### Description

[Meilisearch 0.28.0](https://github.com/meilisearch/meilisearch) / [meilisearch-rails 0.7.0](https://github.com/meilisearch/meilisearch-rails) have introduced breaking API changes to results.raw_answer['nbHits'] used in https://github.com/ddnexus/pagy/blob/master/lib/pagy/extras/meilisearch.rb line 32 & 48

The immediate fix seems to be to replace 'nbHits' with 'estimatedTotalHits' but the Meilisearch documentation does recommend against using this method in general for pagination.

### Steps to reproduce the issue using the provided support

Meilisearch will need to be [installed](https://docs.meilisearch.com/learn/getting_started/quick_start.html#local-installation)/running locally on http://localhost:7700
ruby pagy_bare_rails.rb

Test will fail capturing the Pagy::VariableError - ERROR: expected :count >= 0; got nil

If you wish to remove the 'Record' index created in your local Meilisearch instance then run `curl -X DELETE 'http://localhost:7700/indexes/Record'`

### Edited pagy_bare_rails.rb
```
# frozen_string_literal: true

# Self-contained, bare rails app, usable to play with pagy
# and/or easily reproduce any pagy issue.

# Edit this file as you need

# USAGE:
#     ruby pagy_bare_rails.rb

# NOTICE: if you get any installation error(s) with the following setup
# temporarily remove `pagy/Gemfile` and `pagy/Gemfile.lock` from the repository
# because they may interfere with the bundler/inline.

require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'

  # Debuggers
  # gem 'debug'

  # gem 'debase'
  # gem 'ruby-debug-ide'

  gem 'pagy'            # <--install from rubygems
  # gem 'pagy', path: '../' # <-- use the local repo
  gem 'meilisearch-rails'
  gem 'sqlite3'

  # gem "rails", github: "rails/rails", branch: "main"
  gem 'rails', '~> 6.1'
  gem 'actionpack'
  gem 'railties'
end

# Set up debugging tools
# require "debug" ## etc

# pagy initializer - add your requirements here:
# https://ddnexus.github.io/pagy/extras
# https://ddnexus.github.io/pagy/api/backend

# Edit this section as you need
require 'pagy/extras/metadata'
require 'pagy/extras/overflow'
require 'pagy/extras/trim'
require 'pagy/extras/meilisearch'

Pagy::DEFAULT[:trim_extra] = false
Pagy::DEFAULT[:items] = 20
Pagy::DEFAULT[:metadata] = %i[first_url last_url]
Pagy::DEFAULT[:overflow] = :empty_page


MeiliSearch::Rails.configuration = {
  meilisearch_host: 'http://localhost:7700',
  meilisearch_api_key: nil,
  timeout: 2,
  max_retries: 1,
}

# Rails set up:
require 'action_controller/railtie'
require 'active_record'

require 'minitest/autorun' # runs tests automatically

class TestApp < Rails::Application # :nodoc:
  config.root = __dir__
  config.hosts << 'www.example.com'
  config.hosts << 'example.org'
  config.session_store :cookie_store, key: 'cookie_store_key'
  secrets.secret_key_base = 'secret_key_base'

  ActiveRecord::Base.establish_connection('sqlite3::memory:')
  ActiveRecord::Base.connection.execute('DROP TABLE IF EXISTS records;')
  ActiveRecord::Base.connection.execute('CREATE TABLE records (id int);')

  config.logger = Logger.new($stdout)
  Rails.logger  = config.logger

  routes.draw do
    get '/' => 'test#index', as: 'test'
  end
end

class Record < ActiveRecord::Base
  extend Pagy::Meilisearch
  include MeiliSearch::Rails

  meilisearch do
    attribute :id
  end
end # :nodoc:

class TestController < ActionController::Base # :nodoc:
  include Rails.application.routes.url_helpers

  include Pagy::Backend

  def index
    begin
      results = Record.pagy_search(params[:q])
      pagy, results = pagy_meilisearch(results, items: 10)
    rescue Pagy::VariableError => variable_error
      error = "ERROR: #{variable_error}"
    end
    meta, _records = pagy(Record.none, items: 10)
    render json: { data: [], meta: pagy_metadata(meta), error: error }
  end

  ## override pagy methods if you need to here:
  # def pagy_get_items(collection, pagy)
  #  collection.limit(1)
  # end
end

class TestControllerTest < ActionDispatch::IntegrationTest # :nodoc:
  # Request helpers: https://api.rubyonrails.org/v6.1.4/classes/ActionDispatch/Integration/RequestHelpers.html
  # Available assertions: https://guides.rubyonrails.org/testing.html#available-assertions
  # Rails assertions: https://guides.rubyonrails.org/testing.html#rails-specific-assertions

  def test_my_pagy_problem_here
    get '/'
    assert_equal response.parsed_body, { 'data' => [], 'meta' => { 'first_url' => '/?page=1', 'last_url' => '/?page=1' }, 'error' => nil }
  end

  private

  def app
    Rails.application
  end
end

# or use rack to test:
class BugTest < Minitest::Test # :nodoc:
  include Rack::Test::Methods

  def test_my_pagy_problem_again
    get '/'
    assert last_response.ok?
  end

  private

  def app
    Rails.application
  end
end
```

**Comment by ddnexus** [comment]
@benbws See specific implementation discussion #397

**Comment by jmarsh24** [comment]
Is there a workaround for this issue until it's patched?

**Comment by benbws** [comment]
@jmarsh24 Unfortunately unless you want to use my fork there is currently not a workaround as far as I can tell using the latest versions of Meilisearch and pagy.

**Comment by brunoocasali** [comment]
@jmarsh24 and @benbws, I'll release a new version of the ruby gem tomorrow that will fix this issue. In fact, you can try right now https://github.com/meilisearch/meilisearch-ruby/pull/349

**Comment by ddnexus** [comment]
No need to fix anything after #397
